bentoml/yatai/client/bento_repository_api.py

```diff
@@ -534,6 +629,74 @@ def load(self, bento):
             saved_bundle_path = bento
         else:
             bento_pb = self.get(bento)
-            saved_bundle_path = resolve_bento_bundle_uri(bento_pb)
+            if bento_pb.uri.type == BentoUri.LOCAL and isinstance(
+                self.yatai_service, YataiStub
+            ):
+                saved_bundle_path = self.download_bento(bento_pb.name, bento_pb.version)
+            else:
+                saved_bundle_path = resolve_bento_bundle_uri(bento_pb)
         svc = load_from_dir(saved_bundle_path)
         return svc
+
+    def upload_bento(self, bento_name, bento_version, saved_bento_bundle_path):
+        with TempDirectory() as tarfile_dir:
+            try:
+                tarfile_path, _ = archive_directory_to_tar(
+                    saved_bento_bundle_path, tarfile_dir, bento_version
+                )
+                result = self.yatai_service.UploadBento(
+                    iter(
+                        BentoUploadStreamingRequests(
+                            bento_name, bento_version, tarfile_path
+                        ),
+                    )
+                )
+                if result.status.status_code != status_pb2.Status.OK:
```

Comment:
Does GRPC come with a default timeout behavior?

Reply:
grpc provides a timeout parameter as part of the request method. That's something we should add.  Good catch

Reply:
Will we timeout here if server does not respond?